### ISSUE TITLE
recurring events parsing

### ISSUE THREAD

**Issue by tupsut** [hegza/memoparsa-rs#2]
Both/all of the current test cases feature a set of annually recurring events (and in the case of alpha, also more frequently recurring events). Parsing these has not been implemented - annually recurring events just get duplicates created for the current year, and others do not get parsed at all (as the months are not marked).

It should be quite easy to handle annual events (creating them for a few years in the future) and weekly/monthly events, but more complicated repetition patterns will need some more elaborate planning.

**Comment by hegza** [comment]
Agreed. Though it will likely need a stateful parser. Maybe something to wrap the current line-by-line implementation. I'm getting a better idea of the possible requirements here. I'll look into it once the library is in actually useful state.
